### ISSUE TITLE
Change metagenotype_list_row to avoid hard-coding ontology names

### ISSUE THREAD

**Issue by kimrutherford** [pombase/canto#1644]
In metagenotypeListRow controller "pathogen_host_interaction_phenotype" is hard-coded.  We should get the ontology names from the config.

Here is where it's hard coded at the moment:

```javascript
$scope.createAnnotationUri = CantoGlobals.curs_root_uri +
    '/feature/metagenotype/annotate/' + $scope.metagenotype.metagenotype_id
   + '/start/pathogen_host_interaction_phenotype/';
```

We should also handle the possibility that there are could be more than one annotation type / ontology that makes sense for meta-genotypes.

Here's what we do in genotypeListRowLinksCtrl:

```javascript
        $scope.matchingAnnotationTypes = [];

        AnnotationTypeConfig.getAll().then(function(response) {
          $scope.matchingAnnotationTypes =
            $.grep(response.data,
                   function(annotationType) {
                     if (annotationType.feature_type !== 'genotype') {
                       return false;
                     }

                     if ($scope.pathogen_host_mode &&
                         genotypePathogenOrHost !== annotationType.feature_subtype) {
                       return false;
                     }

                     return true;
                   });
        });
```

In the template we do:

```html
<div ng-repeat="annotationType in matchingAnnotationTypes">
  <a ng-class="{disabled: genotypeId == null }" ng-if="!read_only_curs"
     href="{{curs_root_uri + '/feature/genotype/annotate/' + genotypeId + '/start/' + annotationType.name + '/'}}">
    Start a {{ annotationType.display_name }} annotation ...
  </a>
</div>
```

**Comment by barnabynorman** [comment]
@kimrutherford I have implemented as suggested, however it now offers a different annotation / link (disease formation phenotype) that does not work, ending with a 404

Is this where the configuration is out-of-date or something bigger wrong?

**Comment by kimrutherford** [comment]
> I have implemented as suggested

Thanks.

> ending with a 404

The `curs_root_uri` wasn't in the scope.  It's fixed now in d36d0fd001bb4f281fcd8b8.
